From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 15 Aug 2023 17:14:29 +0200
Subject: clutter: Check that pointer has coordinates prior to repick

This might fail during initialization, or after first pointer
interaction on a pointer-less (e.g. touch-only) session.

(cherry-picked from commit be3bca01a7b8f2cf9d37f1d1a51a21bf86a78ed9)

Origin: upstream, 44.6, commit:f4cf7e455a6250a6be977d6ddae94d53b11da367
---
 clutter/clutter/clutter-stage.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/clutter/clutter/clutter-stage.c b/clutter/clutter/clutter-stage.c
index 70bc00b..025a851 100644
--- a/clutter/clutter/clutter-stage.c
+++ b/clutter/clutter/clutter-stage.c
@@ -3796,9 +3796,11 @@ void
 clutter_stage_repick_device (ClutterStage       *stage,
                              ClutterInputDevice *device)
 {
-  graphene_point_t point;
+  graphene_point_t point = GRAPHENE_POINT_INIT_ZERO;
+
+  if (!clutter_stage_get_device_coords (stage, device, NULL, &point))
+    return;
 
-  clutter_stage_get_device_coords (stage, device, NULL, &point);
   clutter_stage_pick_and_update_device (stage,
                                         device,
                                         NULL, NULL,
